app/services/document_publishing_service.rb

```diff
@@ -16,7 +16,7 @@ def publish_draft(document)
   def publish(document, review_state)
     document.update!(publication_state: "sending_to_live", review_state: review_state)
     publishing_api.publish(document.content_id, nil, locale: document.locale)
-    document.update!(publication_state: "sent_to_live", change_note: nil, update_type: "major", has_live_version_on_govuk: true)
+    document.update!(edition_number: document.edition_number + 1, publication_state: "sent_to_live", change_note: nil, update_type: "major", has_live_version_on_govuk: true)
```

Comment:
Just wondering if it's correct to increment the edition number before we know for sure that the publish has been successful?  I can image if there was an error publishing the document and the user tried to publish again we'd increment the edition number again which doesn't seem quite right?

Reply:
Good point. This is below `publishing_api.publish` so if it reaches this line we're sure the publishing was successful, otherwise an error would have been raised.

Reply:
Are we incrementing this prematurely?

When we have a user_facing_state of "published" it will have an edition number of 2. Whereas that would still be the first edition until it goes to draft?